src/System.Data.SqlClient/tests/ManualTests/SQL/ConnectionPoolTest/PoolBlockPeriodTest.netcoreapp.cs

```diff
@@ -0,0 +1,205 @@
+        [InlineData("Azure with Never Policy must Disable Blocking", new object[] { "nonexistance.database.windows.net", PoolBlockingPeriod.NeverBlock })]
+        public void TestAzureBlockingPeriod(string description, object[] Params)
+        {
+            SqlConnection.ClearAllPools();
+            string serverName = Params[0] as string;
+            PoolBlockingPeriod? policy = null;
+            if (Params.Length > 1)
+            {
+                policy = (PoolBlockingPeriod)Params[1];
+            }
+
+            string connString = CreateConnectionString(serverName, policy);
+            PoolBlockingPeriodAzureTest(connString, policy);
+        }
+
+        [Theory]
+        [InlineData("NonAzure with Default Policy must Enable blocking", new object[] { "nonexistanceserver" })]
+        [InlineData("NonAzure with Auto Policy must Enable Blocking", new object[] { "nonexistanceserver", PoolBlockingPeriod.Auto })]
+        [InlineData("NonAzure with Always Policy must Enable Blocking", new object[] { "nonexistanceserver", PoolBlockingPeriod.AlwaysBlock })]
+        [InlineData("NonAzure with Never Policy must Disable Blocking", new object[] { "nonexistanceserver", PoolBlockingPeriod.NeverBlock })]
+        [InlineData("NonAzure (which contains azure endpoint - nonexistance.WINDOWS.net) with Default Policy must Enable Blocking", new object[] { "nonexistance.windows.net" })]
+        [InlineData("NonAzure (which contains azure endpoint - nonexistance.database.WINDOWS.net.else) with Default Policy must Enable Blocking", new object[] { "nonexistance.database.windows.net.else" })]
+        public void TestNonAzureBlockingPeriod(string description, object[] Params)
+        {
+            SqlConnection.ClearAllPools();
+            string serverName = Params[0] as string;
+            PoolBlockingPeriod? policy = null;
+
+            if (Params.Length > 1)
+            {
+                policy = (PoolBlockingPeriod)Params[1];
+            }
+
+            string connString = CreateConnectionString(serverName, policy);
+            PoolBlockingPeriodNonAzureTest(connString, policy);
+        }
+
+        [Theory]
+        [InlineData("Test policy with Auto (lowercase)", new object[] { "auto" })]
+        [InlineData("Test policy with Auto (miXedcase)", new object[] { "auTo" })]
+        [InlineData("Test policy with Auto (Pascalcase)", new object[] { "Auto" })]
+        public void TestSetPolicyWithAutoVariations(string description, object[] Params)
+        {
+            SqlConnection.ClearAllPools();
+            string policyString = Params[0] as string;
+
+            string connString = CreateConnectionString(_sampleAzureEndpoint, null) + $";{_policyKeyword}={policyString}";
+            PoolBlockingPeriodAzureTest(connString, PoolBlockingPeriod.Auto);
+        }
+
+        [Theory]
+        [InlineData("Test policy with Always (lowercase)", new object[] { "alwaysblock" })]
```

Comment:
This InlineData could be passed to the test above test. The two tests differ in data. The test Code is the same.